### DISCUSSION TITLE
Hitting rate limit on GetUpdates. Seeking for way forward.

### DISCUSSION THREAD

**Discussion by MrVolot** [aiogram/aiogram#1422] [Q&A] [answered]
I keep getting such errors:
```
Failed to fetch updates - TelegramRetryAfter: Telegram server says - Flood control exceeded on method 'GetUpdates'. Retry in 5 seconds.
Original description: Too Many Requests: retry after 5
(background on this error at: https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this)
```

I know that the best option to avoid this would be to set up the webhook. However, I was wondering if there is any other way to avoid hitting rate limits and still use long polling?

Thanks in advance for the input and feedback.

**Comment by JrooTJunior** [comment, marked as answer]
You should do nothing with this error.

`getUpdates` method has no actual rate-limit. This error means that the Telegram Bot API Server is restarting and request should be retried in 5 seconds and this operation is already handled by aiogram itself in the same way as any other network issues - retrying request and notifined in the logs to helps to understand why your bot is not handling new events until server will be still alive.

https://github.com/aiogram/aiogram/blob/a37deff3961d53e7b3a3738c2dd16ee44116938b/aiogram/dispatcher/dispatcher.py#L213-L239
